### PR TITLE
Tabulation documentation

### PR DESCRIPTION
**Pull request by spencer-skylight**
# Description
This PR updates the tabulation package with README and tutorial documentation.

## Additional Changes
This also updates the `phdi/examples/FHIR-to-table` example use case to bring it in line with changes to the PHDI library.

### CODE REVIEW

File: tutorials/tabulation-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Tutorials: The Tabulation Module
+
+This guide serves as a tutorial overview of the functionality available in both `phdi.tabulation` and `phdi.fhir.tabulation`. It will cover concepts such as schemas, FHIR data extraction, and writing tabular data.
+
+## Module Overview
+A single data platform cannot hope to be all things for all people. Simply put, there are fundamental tradeoffs when choosing any data platform, and it is sometimes helpful to move data from one platform to another in order to accomplish a task more efficiently. The PHDI tabulation modules seek to address this problem by defining a data export schema and tools that can map data from one datastore into a tabular format. The tabularized data can then be analyzed using a variety of query and data analysis tools.
```
Comment (bamader) [inline tutorials/tabulation-tutorial.md:6]: Nothing to change, but _love_ this first sentence as an intro to this paragraph!

**Review by bamader — COMMENTED** [review]
The content that's here is excellent. My only question is do we feel it sufficiently covers the functions included in `phdi.tabulation` and `phdi.fhir.tabulation` (not a leading question btw, it's me genuinely wondering)? I love the explanation of a schema and the structure therein, but the only function we explain and show a use case for is `generate_all_tables_in_schema` from `fhir.tabulation`. In `phdi.tabulation`, there's still `load_schema`, `write_table`, and `print_schema summary`; and in `phdi.fhir.tabulation`, there's `apply_selection_criteria`, `apply_schema_to_resource`, and `generate_table`. The ones in fhir.tabulation do strike me somewhat as "under the hood" functions that a user may not need explicit details on because they're used so extensively in generating all tables, but do we think it's worth having some short documentation on some of the functions in the base phdi.tabulation content?

**Comment by spencer-skylight** [comment]
@bamader Thanks for pointing these out! Thinking about it more, I think there are some additions and changes we should make:
* In `phdi.tabulation`, `write_table` and `print_schema_summary`: We should probably document these as well.
* In `phdi.fhir.tabulation`, `generate_table` and `apply_schema_to_resource`: I agree we should include these functions.  
* In `phdi.fhir.tabulation`, `apply_selection_criteria`: I'm thinking we should make this private, and omit from documentation. 

Just let me know what you think, and I'll work on these changes today.  Thanks!

**Comment by bamader** [comment]
I agree with all your thoughts, Spencer. I'm honestly surprised we didn't make `apply_selection_criteria_to_resource` a private underscore function before. The functions you proposed to document sound good to me, though!

**Review by bamader — APPROVED** [review]
Looks great, Spencer! Appreciate the updates.

**Pull request merged**